src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToRfc3339Transformer.php

```diff
@@ -32,15 +39,25 @@ public function transform($dateTime)
         }
 
         if ($this->inputTimezone !== $this->outputTimezone) {
-            $dateTime = clone $dateTime;
+            if (!$dateTime instanceof \DateTimeImmutable) {
+                $dateTime = clone $dateTime;
+            }
```

Comment:
The point is that the `DataTransformerInterface` just states that the given input should be transformed and then being returned. To me this does not include that the input is allowed to change.